Sophia Sun Project Pitch

```diff
@@ -0,0 +1,244 @@
+the chord.
+
+Overall, I think the language component seems much smaller than the "systems" aspect. I think the language also has the 
+possibility for extensibilityby allowing the user to indicate the time signature, length of the note, and rests 
+to create a very simple score. However, this would probably end up adding even more to the systems aspect. 
+
+
+
+# More information about the needs user and about the suitability of the idea as a CS 111 project (all the angles, the benefits and drawbacks, risks, etc.)
++ More benefits: 
++ I think it really fits the skill level and knowledge base of my users (and there are a lot of possible users!) 
++ I can ask an outside software, such as LilyPond, to do the difficult work of displaying the notes on a sheet music.
+
++ More drawbacks/risks:
++ The DSL requires some kind of communication with another soft 
++ If i want the error messages to be very understandable and helpful, I might need to anticipate a lot of different types of errors.
++ At this point, I'm not very familiar with LilyPond, so I might not be able to anticipate all the errors that occur when my DSL communicates with it. That means that i might not always be able to provide the best error messages.
+
+
+# Challenges of the project and my current open questions about the project
++ Challenges: communicating my DSL to LilyPond. I'm not that familiar with LilyPond, and I don't know how a DSL can communicate with it, so I will have to put in a lot of work to understand how LilyPond words and fit my DSL to it.
++ Open question: what kinds of software does the user need to have installed on his/her computer for this to work??
+
+# Initial, well-reasoned speculation about how the implementation will proceed, including: what you plan to work on first, whether the project might work best as an internal or external language, candidates for potential host languages, etc.
++ What I plan to work on first: the basic features are outlined in this pitch.
++ Whether the project might work best as an internal or external language: I think the project might work best as an external language, because I want the user input to be as abstracted from the 
+parsing and intermediate representations as possible. This is because the user of the language is someone who has found a chord sheet and wants to play it asap, and thus needs a quick and easy way to translate the chords he/she finds into sheet music. I also expect the user to be a music fan who can read sheet music, but not enough theory to quickly read chord sheets. Thus, I want to keep the program as simple as possible.
++ Candidates for potential host languages: I think the best language to use would be Scala? Through the Piconot lab, I learned how to read in lines of a txt file and to parse a string. I think the input of my project is similar to the input of the Piconot lab, and thus I might use Scala. I'm also not familiar with how to use other languages, which is why I'm choosing Scala!
```

Comment:
Prof Ben says that Scala has better parsing tools, and that the learning curve for Java parsing is okay until a certain point where it suddenly becomes more difficult.